crypto/crypto.go

```diff
@@ -55,7 +61,23 @@ func (c crypto) CreateQuorumCert(block *hotstuff.Block, signatures []hotstuff.Pa
 	if err != nil {
 		return hotstuff.QuorumCert{}, err
 	}
-	return hotstuff.NewQuorumCert(sig, block.View(), block.Hash()), nil
+	latencyVector := c.prepareLatencyVector(block, signatures)
+	return hotstuff.NewQuorumCert(c.opts.ID(), sig, block.View(), block.Hash(), latencyVector), nil
+}
+
+func (c crypto) prepareLatencyVector(block *hotstuff.Block, signatures []hotstuff.PartialCert) map[uint32]uint64 {
+	latencyVector := make(map[uint32]uint64)
```

Comment:
This is a general comment that applies in many places. Here you are allocating memory for the map and if you return early below, the map is empty. I think this may cause some unnecessary memory pressure and slow things down unnecessarily. I think it should be fine to return `nil` here (and other places). If you later try to iterate over a `nil` map, it will be as if it was an empty map.

You may need to add some `nil` guards in other places, but didn't check.